sklearn/utils/validation.py

```diff
@@ -353,6 +353,13 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         The converted and validated X.
 
     """
+
+    # accept masked check
+    masked = hasattr(array,'mask')
+    if not accept_masked and masked:
+        raise TypeError('Masked arrays are not supported.')
+    mask = False if not masked else array.mask
```

Comment:
np.ma.getmask(array)

Reply:
Are you referring just to line 361 or the whole thing?

Reply:
Just this line